org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/launch/DefaultGradleLaunchConfigurationManager.java

```diff
@@ -74,7 +75,13 @@ private ILaunchConfiguration createLaunchConfiguration(GradleRunConfigurationAtt
             configurationAttributes.apply(launchConfiguration);
 
             // persist the launch configuration and return it
-            return launchConfiguration.doSave();
+            try {
```

Comment:
⭕️ A minor readability issue: instead of nested try-catch blocks, please extract the `doSave()` call to a new `persistConfiguration` method.